### DISCUSSION TITLE
[Headless] have more feedback about the execution of Gama in headless mode

### DISCUSSION THREAD

**Discussion by chapuisk** [gama-platform/gama#64] [Ideas]
**Is your request related to a problem? Please describe.**
When launching Gama on a server, we are quite in the dark, meaning that we have little feedback on the execution of the program; except for errors. For instance, I suspect Gama to produce memory leaks when saving in files within models, but there is no straightforward ways to inspect Gama execution. We can have information about simulations launched using write statement or even log files, but we cannot log memory usage, CPU usage and the like from the JVM. 

**Describe the improvement you'd like**
To have an option in Gama.ini or in the headless batch script that allows to track Gama execution (either a full stack or target by target, i.e. RAM or CPU usage over time.

**Additional context**
I try to launch headless on a HPC but have very few feedback from the host and even less from Gama; hence it is very difficult to understand why it fails and how to fix execution, especially when it stops during an experiment

**Comment by AlexisDrogoul** [comment]
Hi,

I think that what you are asking for is clearly beyond the scope of GAMA itself. The exact same problem arises when running any Java program, remotely or not. There are tools to do what you describe, but they are deeply integrated with the JVM and/or the OS, not with any particular program like GAMA. See for instance how you can use a combination of visualvm and JMX to remotely debug / monitor / profile a Java app here : https://stackoverflow.com/questions/30069643/remote-monitoring-with-visualvm-and-jmx

JProfile, YourKit, etc. also allow to do remote profiling or monitoring at various degrees and this is what you should be looking for. But GAMA will never provide such a service by itself 

For this reason, and in order to gather other feedbacks, I will move this thread to the "discussions" -- and maybe you should duplicate it to the mailing list ?
